site/book/04-using-functions/02-imperative-function-execution.md

```diff
@@ -1 +1,174 @@
-TODO
+The `fn eval` command enables you to execute a single function without declaring it in the package.
+This is referred to as imperative function execution.
+
+For example, to set the namespace of all resources in the wordpress package hierarchy:
+
+```shell
+$ kpt fn eval --image gcr.io/kpt-fn/set-namespace:v0.1 wordpress -- namespace=mywordpress
+```
+
+?> Refer to the [command reference][eval-doc] for usage.
+
+This changes the resources in the `wordpress` package and the `mysql` subpackage.
+
+For comparison, this has the same effect as the following declaration:
+
+```yaml
+# wordpress/Kptfile (Excerpt)
+apiVersion: kpt.dev/v1alpha2
+kind: Kptfile
+metadata:
+  name: wordpress
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/set-namespace:v0.1
+      configMap:
+        namespace: mywordpress
+```
+
+So when should you execute a function using `eval` instead of `render`?
+
+When you have one of these use cases:
+
+- Perform a one-time operation
+- Execute a function from a CI/CD system on packages authored by other teams
+- Develop shell scripts and chain functions with Unix pipeline (`|`)
+- Mutate meta resources such as the `Kptfile` (Not allowed by `render`)
+- Execute the function with privilege (Not allowed by `render`)
+
+We will cover these topics in detail.
+
+## Specifying `functionConfig`
+
+There are two ways to specify the `functionConfig`.
+
+### `fn-config` flag
+
+The general way to provide a `functionConfig` of arbitrary kind (core or custom resources), is to
+declare the resource in a separate file and use the `fn-config` flag.
+
+```shell
+$ cat << EOF > /tmp/fn-config.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ns
+data:
+  namespace: mywordpress
+EOF
+```
+
+```shell
+$ kpt fn eval --image gcr.io/kpt-fn/set-namespace:v0.1 --fn-config /tmp/fn-config.yaml wordpress
```

Comment:
Shouldn't flags come at the end after DIR as per pattern? https://kpt.dev/reference/

Reply:
go/kpt-docs-style-guide D5

I also updated that reference doc.